pylearn2/datasets/dataset.py

```diff
@@ -48,6 +48,28 @@ def iterator(self, mode=None, batch_size=None, num_batches=None,
             through the dataset and may potentially be shared by
             multiple iterator objects simultaneously (see "Notes"
             below).
+        data_specs : (space, source) pair, optional
+            `space` is an instance of `Space` (possibly a
+            `CompositeSpace`), and `source` is a string (or tuple of
+            strings if `space` is a `CompositeSpace`). `source` defines
+            which data should be returned by the iterator (for instance
+            'features', 'targets'), and `space` specifies in which
+            format it will be returned. These specifications should be
+            flat: if `space` is a `CompositeSpace`, the sub-spaces should
```

Comment:
What are the contexts where the space does have to be CompositeSpace? I remember encountering error messages relating to that while implementing models that only require a single variable. For example, sgd.py and monitor.py seem to require that the root be CompositeSpace. Are there other cases like that, other than things that are obviously intrinsically composite like CompositeLayer?

If Dataset.iterator accepts non-CompositeSpace roots, why do the other contexts have that restriction?